src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -767,8 +767,16 @@ trait SearchConverterService {
                                              taxonomyType: String,
                                              bundle: TaxonomyBundle): Try[List[SearchableTaxonomyContext]] = {
       val (resources, topics) = getTaxonomyResourceAndTopicsForId(id, bundle, taxonomyType)
-      val resourceContexts = resources.map(resource => getResourceTaxonomyContexts(resource, bundle))
-      val topicContexts = topics.map(topic => getTopicTaxonomyContexts(topic, bundle))
+      val resourceContexts = resources
+        .filter(resource =>
+          bundle.getSubject(resource.path).exists(subject => subject.metadata.exists(metadata => metadata.visible)))
+        .filter(resource => resource.metadata.exists(metadata => metadata.visible))
```

Comment:
Sjekker først om subject er visible. Så sjekkes elementet for det samme. Usynlig subject vil alltid trumfe elementet.